upcloud/network.go

```diff
@@ -88,15 +88,27 @@ func (s *Interface) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type DHCPRoutesConfig struct {
+	EffectiveRoutesAutoPopulation EffectiveRoutesAutoPopulation `json:"effective_routes_auto_population"`
+}
+
+type EffectiveRoutesAutoPopulation struct {
+	Enabled             Boolean  `json:"enabled"`
+	ExcludeBySource     []string `json:"exclude_by_source,omitempty"`
+	FilterByDestination []string `json:"filter_by_destination,omitempty"`
+	FilterByRouteType   []string `json:"filter_by_route_type,omitempty"`
```

Comment:
We have in some cases (e.g. `type RouterStaticRouteType string` [here](https://github.com/UpCloudLtd/upcloud-go-api/blob/main/upcloud/network.go#L5)) used custom string types for string where there are known possible values, like we have `router-connected-networks` and/or `static-route` for `exclude_by_source` as well as `user` and/or `service` for `filter_by_route_type`, but have not done that for `family` (that only allows `IPv4`) in networks so I guess either would work here 🤔